kotlin-analysis-core/src/main/kotlin/org/jetbrains/research/ml/kotlinAnalysis/Analyzers.kt

```diff
@@ -0,0 +1,119 @@
+package org.jetbrains.research.ml.kotlinAnalysis
+
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiRecursiveElementVisitor
+
+typealias AnalyzerToStat<P, R> = Map<PsiSecondaryAnalyzer<P, R>, MutableMap<P, R>>
+
+/**
+ * Classes that inherit from this interface implement secondary analyzers that analyze the concrete type of PSI elements.
+ *
+ * @param P the type of PSI elements, that can be analyzed.
+ * @param R the type of the analysis result.
+ */
+interface PsiSecondaryAnalyzer<P : PsiElement, R> {
+    /**
+     * Returns the analysis result of a [PSI element][psiElement].
```

Comment:
Replace into `Analyzes a psi element and returns the analysis result, for example, get a full qualified name for a class or function`
since it is unclear what you mean [PSI element][psiElement]